rainier-core/src/main/scala/com/stripe/rainier/compute/Encoder.scala

```diff
@@ -0,0 +1,70 @@
+package com.stripe.rainier.compute
+
+trait Encoder[T] {
+  type U
+  def wrap(t: T): U
+  def create(acc: List[Variable]): (U, List[Variable])
+  def extract(t: T, acc: List[Double]): List[Double]
+}
+
+object Encoder {
+  type Aux[X, Y] = Encoder[X] { type U = Y }
+  implicit val int: Aux[Int, Real] =
```

Comment:
Does the `: Aux[Int, Real]` help with type inference in other places? Is there something weird about that since `U` is a type member, the compiler doesn't know both `Int` and `U=Real` at the same time without it?

Reply:
No, it doesn't affect anything other than how concisely I can express the annotations.